kernel/thread.c

```diff
@@ -248,6 +248,37 @@ int z_impl_k_thread_name_copy(k_tid_t thread_id, char *buf, size_t size)
 #endif /* CONFIG_THREAD_NAME */
 }
 
+const char *k_thread_state_str(k_tid_t thread_id)
+{
+	switch (thread_id->base.thread_state) {
```

Comment:
think this is fundamentally flawed, although it might work most of the time.
better to examine each bit at a time.

Reply:
but won't it create more confusion if by any chance shell output indicates that thread is in two states? Isn't it better to print `unknown` or `in transition` for that very unlikely (to be read out in low priority thread) transient state? at least state naming suggests that states are exclusive.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/77db273f6f84b3d94d34b5e72070ec6bc31e41cf/lib/cmsis_rtos_v2/thread.c#L302
```	
        case _THREAD_DUMMY:
		state = osThreadError;
		break;
	case _THREAD_PRESTART:
		state = osThreadInactive;
		break;
```
Treats thread state as exclusive. 
Suppose, we need to get rid of this implicity in thread states (smth. like simple comment in header file)

Reply:
Since this is the main contention point, can @andyross please weigh in?

Reply:
fine w/this as-is